app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1191,6 +1234,209 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
         }
     }
 
+    inner class BrowserTabFragmentExperimentDecorator {
+        fun decorateToolbarMenus(viewState: BrowserViewState) {
+            if (variantManager.getVariant().hasFeature(VariantManager.VariantFeature.BottomBarNavigationExperiment)) {
+                decorator.hideToolbarMenu()
+            } else {
+                decorator.decorateToolbarMenu(viewState)
+            }
+        }
+
+        private fun decorateToolbarMenu(viewState: BrowserViewState) {
+            tabsButton?.isVisible = viewState.showTabsButton
+            fireMenuButton?.isVisible = viewState.showFireButton
+            menuButton?.isVisible = viewState.showMenuButton
+        }
+
+        private fun hideToolbarMenu() {
+            menuButton?.gone()
+        }
+
+        fun decorateWithToolbarOnlyExperiment() {
+            hideBottomBar()
+            decorateAppBarWithToolbarOnlyExperiment()
+            createPopupMenuWithToolbarOnlyExperiment()
```

Comment:
it is, not sure I understand what do you mean?

Reply:
I got confused because `decorateAppBarWithToolbarOnlyExperiment`only contains logic to:
1. add tab menu option + fire menu option
2. configure fire button click listener

I was expecting more logic inside that method, kind of to fully configure the appbar, including the popupmenu and other tab switcher listener.